addons/im_livechat/models/im_livechat_channel.py

```diff
@@ -162,8 +162,10 @@ def _get_mail_channel(self, anonymous_name, previous_operator_id=None, user_id=N
         operator_partner_id = operator.partner_id.id
         # partner to add to the mail.channel
         channel_partner_to_add = [(4, operator_partner_id)]
-        if self.env.user and self.env.user.active:  # valid session user (not public)
-            channel_partner_to_add.append((4, self.env.user.partner_id.id))
+        if user_id:
+            current_user = self.env['res.users'].browse(user_id)
```

Comment:
This is visitor's user, not necessarily current user, hence I would name it `visitor_user` instead.

Reply:
@alexkuhn done, I've changed it to `visitor_user`